crates/pathfinder/src/p2p_network/sync_handlers/v1/tests.rs

```diff
@@ -0,0 +1,554 @@
+use p2p_proto_v1::common::{Direction, Step};
+use pathfinder_common::BlockNumber;
+use rstest::rstest;
+
+const I64_MAX: u64 = i64::MAX as u64;
+
+#[rstest]
+#[case(0, 1, Direction::Forward, Some(1))]
+#[case(0, I64_MAX, Direction::Forward, Some(I64_MAX))]
+#[case(1, I64_MAX, Direction::Forward, None)]
+#[case(0, 1, Direction::Backward, None)]
+#[case(1, 1, Direction::Backward, Some(0))]
+#[case(I64_MAX, 1, Direction::Backward, Some(I64_MAX - 1))]
+#[case(I64_MAX, I64_MAX, Direction::Backward, Some(0))]
+#[case(I64_MAX, I64_MAX + 1, Direction::Backward, None)]
+#[test]
+fn get_next_block_number(
+    #[case] start: u64,
+    #[case] step: u64,
+    #[case] direction: Direction,
+    #[case] expected: Option<u64>,
+) {
+    assert_eq!(
+        super::get_next_block_number(
+            BlockNumber::new_or_panic(start),
+            Step::from(Some(step)),
+            direction
+        ),
+        expected.map(BlockNumber::new_or_panic)
+    );
+}
+
+mod empty_reply {
```

Comment:
As of current spec all of the handlers should react in the same way in border cases.